packages/client/verifiable-credentials/src/presentation/getNfts.ts

```diff
@@ -35,13 +34,36 @@ export async function getWalletNfts(chain: string, wallet: string, environment:
             }
         } catch (error) {
             console.error(error);
-            return null;
+            throw new Error(`Failed to get nfts`);
         }
     }
 
     return allData;
 }
 
-export function filterPolygonErc721(nfts: VC_EVMNFT[]): VC_EVMNFT[] {
-    return nfts.filter((nft) => isPolygon(nft.chain) && nft.tokenStandard === "erc-721");
+export function filterPolygonErc721(nfts: CrossmintWalletNft[]): VCNFT[] {
+    const vcNfts: VCNFT[] = [];
+    for (const nft of nfts) {
+        if (isVcChain(nft.chain) && nft.tokenStandard === "erc-721") {
+            vcNfts.push({
+                chain: nft.chain,
+                contractAddress: nft.contractAddress,
+                tokenId: nft.tokenId,
+            });
+        }
+    }
+    return vcNfts;
+}
+
+export async function getWalletVcCompatibleNfts(chain: string, wallet: string): Promise<VCNFT[]> {
```

Comment:
This logic has been extracted from the old getCredentialsCollection method, no code changes